src/member-api.js

```diff
@@ -163,10 +163,10 @@ export class MemberApi extends TypedEmitter {
 
   /**
    * @param {string} deviceId
-   * @param {import('./capabilities.js').RoleId} roleId
+   * @param {import('./roles.js').RoleIdAssignableToOthers} roleId
    * @returns {Promise<void>}
    */
   async assignRole(deviceId, roleId) {
```

Comment:
Can we add a TODO note here or open an issue to have stricter runtime checks here.
I also wonder if "assigning a blocked role" is confusing for the API consumer, and we should expose that separately as `$member.block(deviceId)` and limit `assignRole()` to just assignable roles within the project (currently participant and coordinator).

Reply:
Also in the future we might want to add options to how we block users (which internally we might implement as different roles, or additional fields on membership records - maybe the latter better for many scenarios), for example:

1. Removed with "self-destruct", e.g. when a device discovers it is removed it deletes all data including its own. Would be useful for compromised phones.
2. Removed with removed encryption keys and keeping own data (but deleting others) - useful for when removing a device that we know has important data on it, that we want to sync off it in the future, without allowing that device to access it. Device would need to self-delete its encryption keys when it finds out it is blocked this way.
3. Removed with ignore all data from device - would re-index and ignore all edits or created data from the removed device.

Having a separate API to remove devices would be helpful for implementing these.

Reply:
Added a stricter runtime check in f24c85e3d807ce58e3f4b99a148f6ee1c9d345b6. Let me know if you still want me to add a TODO/issue.